ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -24,6 +24,95 @@
 /* Byte size of CKA_ID attribute when generated locally */
 #define PKCS11_CKA_DEFAULT_SIZE		16
 
+static uint32_t pkcs11_func2ckfm(enum processing_func function)
+{
+	switch (function) {
+	case PKCS11_FUNCTION_DIGEST:
+		return PKCS11_CKFM_DIGEST;
+	case PKCS11_FUNCTION_GENERATE:
+		return PKCS11_CKFM_GENERATE;
+	case PKCS11_FUNCTION_GENERATE_PAIR:
+		return PKCS11_CKFM_GENERATE_KEY_PAIR;
+	case PKCS11_FUNCTION_DERIVE:
+		return PKCS11_CKFM_DERIVE;
+	case PKCS11_FUNCTION_WRAP:
+		return PKCS11_CKFM_WRAP;
+	case PKCS11_FUNCTION_UNWRAP:
+		return PKCS11_CKFM_UNWRAP;
+	case PKCS11_FUNCTION_ENCRYPT:
+		return PKCS11_CKFM_ENCRYPT;
+	case PKCS11_FUNCTION_DECRYPT:
+		return PKCS11_CKFM_DECRYPT;
+	case PKCS11_FUNCTION_SIGN:
+		return PKCS11_CKFM_SIGN;
+	case PKCS11_FUNCTION_VERIFY:
+		return PKCS11_CKFM_VERIFY;
+	case PKCS11_FUNCTION_SIGN_RECOVER:
+		return PKCS11_CKFM_SIGN_RECOVER;
+	case PKCS11_FUNCTION_VERIFY_RECOVER:
+		return PKCS11_CKFM_VERIFY_RECOVER;
+	default:
+		return 0;
+	}
+}
+
+enum pkcs11_rc
+check_mechanism_against_processing(struct pkcs11_session *session,
+				   enum pkcs11_mechanism_id mechanism_type,
+				   enum processing_func function,
+				   enum processing_step step)
+{
+	bool allowed = false;
+
+	switch (step) {
+	case PKCS11_FUNC_STEP_INIT:
+		switch (function) {
+		case PKCS11_FUNCTION_IMPORT:
+		case PKCS11_FUNCTION_COPY:
+		case PKCS11_FUNCTION_MODIFY:
+		case PKCS11_FUNCTION_DESTROY:
+			return PKCS11_CKR_OK;
+		default:
+			break;
+		}
+		allowed = mechanism_supported_flags(mechanism_type) &&
```

Comment:
I think here it should be `&`, not `&&`: mechanism supports the target function.

Reply:
Agree.